pvlib/tools.py

```diff
@@ -425,3 +429,25 @@ def _golden_sect_DataFrame(params, VL, VH, func):
             raise Exception("EXCEPTION:iterations exceeded maximum (50)")
 
     return func(df, 'V1'), df['V1']
+
+
+def datetime_to_julian(times):
```

Comment:
I don't like the duplication of `datetime_to_djd`. Let's make the old function work with both `datetime` and `DatetimeIndex`. I suspect converting a `datetime` input to a `Timestamp` will do the job.